api/v1alpha1/stardoguser_types.go

```diff
@@ -25,17 +25,29 @@ import (
 
 // StardogUserSpec defines the desired state of StardogUser
 type StardogUserSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// StardogInstanceRef describes a Stardog instance
+	StardogInstanceRef string `json:"stardogInstanceRef,omitempty"`
+	// StardogUserCredentialsSpec describes the credentials of a Stardog user
+	Credentials StardogUserCredentialsSpec `json:"credentials,omitempty"`
+	// Roles describe a list of StardogRoles assigned to a Stardog user. The names are referring the StardogRole metadata names, not the role name that is supposed to be in Stardog.
+	Roles []string `json:"roles,omitempty"`
+}
 
-	// Foo is an example field of StardogUser. Edit StardogUser_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+// StardogUserCredentialsSpec specifies the password of a Stardog user
+type StardogUserCredentialsSpec struct {
+	// Namespace specifies the namespace of the secret.
+	// Defaults to metadata.namespace
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Where is this `+optional` coming from resp. why is it needed? AFAIK every field is optional, it's more of a question what the operator does with empty values.
```suggestion
	// Namespace specifies the namespace of the Secret referenced in SecretRef.
	// Defaults to .metadata.namespace.
	Namespace string `json:"namespace,omitempty"`
```

Reply:
Is the _optional_ making the field not mandatory?